apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -17,9 +17,33 @@ package v1alpha1
 import (
 	autoscalingv2 "k8s.io/api/autoscaling/v2"
 	v1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// Ingress is used to specify how OpenTelemetry Collector is exposed. This
+// functionality is only available if one of the valid modes is set.
+// Valid modes are: deployment, daemonset and statefulset.
+type Ingress struct {
+	// Type default value is: none
+	// Supported types are: ingress
+	// TODO(frzifus): support routes.
+	Type IngressType `json:"type,omitempty"`
+
+	// Hostname by which the ingress proxy can be reached.
+	// +optional
+	Hostname string `json:"hostname,omitempty"`
```

Comment:
Could you please explain this configuration? Why is it needed? 

https://kubernetes.io/docs/concepts/services-networking/ingress/


PS: I don't have much experience with ingress so I might ask something stupid here.

Reply:
With the ingress entry we want to direct incoming traffic on port 80 or 443 to our service. Without a set hostname, this applies to all domains, which can not be intentional. For example if traffic with the same path but from another domain should be directed to another service.

It could look like this:
```
NAMESPACE  NAME            HOST/PORT            PATH     SERVICES                  PORT
test123    jaeger-skupper  example.com          /traces  jaeger-skupper-collector  https-collector
test456    jaeger-ingress  another-example.com  /traces  jaeger-ingress-collector  grpc-collector
```